File: backend/app/models/job_description.py
from datetime import datetime
from typing import Any

from sqlalchemy import DateTime, ForeignKey, Text, func
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class JobDescription(Base):
    __tablename__ = "job_descriptions"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    raw_text: Mapped[str] = mapped_column(Text, nullable=False)
    parsed_json: Mapped[dict[str, Any] | None] = mapped_column(JSONB, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

    user = relationship("User", back_populates="job_descriptions")

File: frontend/src/types/jobDescription.ts
export type ParsedJobData = {
  skills: string[];
  technologies: string[];
  responsibilities: string[];
  keywords: string[];
};

export type JobDescription = {
  id: number;
  user_id: number;
  raw_text: string;
  parsed_json: ParsedJobData | null;
  created_at: string;
};
